js/skjema/Inputfelt.js

```diff
@@ -17,66 +27,89 @@ const handleKeyPress = (avbryt, e) => {
   }
 };
 
+const validateInput = (input) => {
+  if (input === '') {
+    return texts.errors.emptyInput;
+  } else if (input.length > nameMaxLen) {
+    return texts.errors.maxLengthExceeded;
+  } else if (input.match(tekstfeltInneholderEllerBegynnerMedUgyldigTegnRegex) || input.match(tekstfeltRegex)) {
+    return texts.errors.invalidCharacters;
+  }
+  return null;
+};
+
 const Inputfelt = (props) => {
-  const { meta, className, input, id, avbryt, oppdateringFeilet, varselTekst, spinner } = props;
-  return (
-    <div className="arbeidsgiveroppgave__rad">
-      <input
-        className={`${className}${meta.touched && meta.error ? ' skjemaelement__input--harFeil' : ''}`}
-        {...input}
-        autoComplete="off"
-        placeholder={props.placeholder}
-        type={props.type || 'text'}
-        id={id}
-      />
-      <div className="arbeidsgiveroppgave__rad--feilmelding">
-        <Feilmelding {...meta} />
-      </div>
+  const { oppdateringFeilet, spinner, onSubmit, avbryt } = props;
+  const [arbeidsoppgaveInputText, setArbeidsoppgaveInputText] = useState('');
+  const [errorMsg, setErrorMsg] = useState('');
+
+  const handleChange = (event) => {
+    const input = event.target.value;
+    setArbeidsoppgaveInputText(input);
+    const error = validateInput(input);
```

Comment:
Vil man nå få feilmeldingene med en gang, med en gang man begynner å skrive? Det er noen litt strenge regler i designsystemet for når feilmeldingene skal oppstå, og hvordan fremvise: https://design.nav.no/designsystem/patterns/form-validation

Reply:
Nei, ikke før man submitter med "Lagre"-knapp. Koden er kanskje litt utydelig, men tanken er at feilteksten skal forsvinne når den rettes opp etter etter en tekstforandring, første submit-forsøk

Reply:
Nei, ikke før man submitter med "Lagre"-knapp. Koden er kanskje litt utydelig, men tanken er at feilteksten skal forsvinne når den rettes opp etter etter en tekstforandring, første submit-forsøk.

Reply:
Fikset litt på koden slik at det kom bedre frem når valideringen kjører 👍